### PR TITLE
add configurable-http-proxy

### PR DESCRIPTION
**Pull request by Dentrax**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: configurable-http-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      mkdir -p ${{targets.destdir}}/srv/${{package.name}}
+      cp chp-docker-entrypoint package.json index.js ${{targets.destdir}}/srv/${{package.name}}/
+      cp -r bin/ doc/ lib/ test/ ${{targets.destdir}}/srv/${{package.name}}/
+
+  - working-directory: ${{targets.destdir}}/srv/${{package.name}}
+    pipeline:
+      - runs: |
+          npm install --package-lock-only
+          npm ci --production
+          npm audit fix --package-lock-only --legacy-peer-deps || true
```
Comment (imjasonh) [inline configurable-http-proxy.yaml:44]: Why `|| true`?

File: configurable-http-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package:
+  name: configurable-http-proxy
+  version: 4.6.0
+  epoch: 0
+  description: "HTTP parser written against llparse"
+  copyright:
+    - license: BSD 3-Clause
```
[the remark this answers is not in the document]
Comment (hectorj2f) [reply configurable-http-proxy.yaml:7]: I am more concern about having packages using `BSD N-Clause` and `BSD-N-Clause`.

File: configurable-http-proxy.yaml (hunk 1 above)
Comment (Dentrax) [reply configurable-http-proxy.yaml:44]: Hmmm, no particular reason for this, copy-paste issue probably. But i think its a nit.

Comment (imjasonh) [inline configurable-http-proxy.yaml:44]:
```suggestion
          npm audit fix --package-lock-only --legacy-peer-deps
```

**Review by imjasonh — DISMISSED** [review]

**Review by imjasonh — APPROVED** [review]

**Pull request merged**